CONTRIBUTING.md

```diff
@@ -1,17 +1,195 @@
+
+- Navigate into the project folder:
+
+- cd forgerock-javascript-sdk
+
+- Build the project to make sure it works, `pnpm build`
+
+### Commands in the repo
+
+Most commands live within the local `project.json` of the package you are working in. You can use the project.json and the Nx vscode extension to run commands via the code lens.
+
+Alternatively there are many ways to run commands, of which a few are to be listed below:
+`pnpm build` will run build on all projects
+`pnpm lint` will run lint on all projects
+`pnpm test` will run unit tests on all projects
+`pnpm e2e` will run e2e tests for all e2e apps
+
+You can run commands with `nx` like so:
+`pnpm nx build <name in project.json>`
+
+_examples_:
+
+```
+`pnpm nx build javascript-sdk`
+`pnpm nx lint javascript-sdk`
+`pnpm nx test javascript-sdk`
+`pnpm nx e2e autoscript-suites`
+`pnpm nx serve autoscript-apps`
+```
+
+## Understanding the Project Structure
+
+The Ping SDK for Javascript is organized in a modular way. This section is designed to help you understand the layout of the project. We will break down each of the folders and what modules you will find there. Getting familiar with the project structure will make contributing easier and more efficient.
+
+```
+ e2e/
+   autoscript-apps
+   autoscript-suites
+   mock-api
+   mock-api-v2
+   token-vault-app
+   token-vault-interceptor
+   token-vault-proxy
+   token-vault-suites
+ packages/
+   javascript-sdk
+   ping-protect
+   token-vault
+
+```
+
+### Package overview
+
+- _javascript-sdk_: This is the core module that powers the rest of the packages. OIDC/AIC Callback support/etc
+- _ping-protect_: This is the core package that allows a developer to interact with the Ping Protect signals api. This module will load the signals sdk for you, and provides a lightweight interface for interacting with the modules.
+- _token-vault_: A propreitary way of managing tokens for environments which require high security.
+
+### E2E Overview
+
+```
+- *autoscript-suites*: The e2e test location for writing modular and journey e2e's with the javascript-sdk or ping-protect.
+- *autoscript-apps*: A multi-page web application for loading javascript files which call through the journey you are testing programatically
+- *mock-api*: A backend server that serves mock data for the autoscript-apps. This is hard-coded data to mimic what the actual AIC Tenant will respond with
+- *mock-api-v2*: A new version of the mock api, development of this is moved over to the new Ping SDK repo.
+- *token-vault-app*: The application which sets up Token-Vault for testing
+- *token-vault-proxy*: The proxy application (required) for using token vault
+- *token-vault-suites*: The e2e test location for playwright tests.
+```
+
+## Changesets
+
+We use changesets for releasing our packages. you can add a changeset when needed (when a change should require a release) using `pnpm changeset`
+
+fill out the TUI information and add this to your PR.
```

Comment:
Dangling sentence?
If it's a follow on from the previous, maybe split it into bullets?